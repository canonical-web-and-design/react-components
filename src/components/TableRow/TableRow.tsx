import PropTypes from "prop-types";
import type { HTMLProps, ReactNode } from "react";
import React from "react";

type Props = {
  children: ReactNode;
} & HTMLProps<HTMLTableRowElement>;

/**
 * TableRow
 *
 * @remarks
 * Implementation of TableRow for Table
 * https://vanillaframework.io/docs/base/tables
 *
 * @param children
 * @returns TableRow
 */
const TableRow = ({ children, ...trProps }: Props): JSX.Element => (
  <tr role="row" {...trProps}>
    {children}
  </tr>
);

TableRow.propTypes = {
  children: PropTypes.node,
};

export default TableRow;
