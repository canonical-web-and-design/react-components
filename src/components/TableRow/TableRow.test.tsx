import { shallow } from "enzyme";
import React from "react";

import TableRow from "./TableRow";

describe("TableRow", () => {
  it("renders", () => {
    const wrapper = shallow(
      <TableRow>
        <td></td>
      </TableRow>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
