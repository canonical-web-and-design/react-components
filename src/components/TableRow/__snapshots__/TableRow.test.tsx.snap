// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow renders 1`] = `
<tr
  role="row"
>
  <td />
</tr>
`;
